cirq/contrib/quantum_volume/quantum_volume.py

```diff
@@ -91,45 +92,116 @@ def compute_heavy_set(circuit: cirq.Circuit) -> List[int]:
     ]
 
 
-def sample_heavy_set(circuit: cirq.Circuit,
+@dataclass
+class CompilationResult:
+    swap_network: ccr.SwapNetwork
+    parity_map: Dict[cirq.Qid, cirq.Qid]
+
+
+def sample_heavy_set(compilation_result: CompilationResult,
                      heavy_set: List[int],
                      *,
                      repetitions=10_000,
-                     sampler: cirq.Sampler = cirq.Simulator(),
-                     mapping: Dict[cirq.ops.Qid, cirq.ops.Qid] = None) -> float:
+                     sampler: cirq.Sampler = cirq.Simulator()) -> float:
     """Run a sampler over the given circuit and compute the percentage of its
        outputs that are in the heavy set.
 
     Args:
-        circuit: The circuit to sample.
+        compilation_result: All the information from the compilation.
         heavy_set: The previously-computed heavy set for the given circuit.
         repetitions: The number of times to sample the circuit.
         sampler: The sampler to run on the given circuit.
-        mapping: An optional mapping from compiled qubits to original qubits,
-            to maintain the ordering between the model and compiled circuits.
 
     Returns:
         A probability percentage, from 0 to 1, representing how many of the
         output bit-strings were in the heavy set.
 
     """
+    mapping = compilation_result.swap_network.final_mapping()
+    circuit = compilation_result.swap_network.circuit
+
     # Add measure gates to the end of (a copy of) the circuit. Ensure that those
     # gates measure those in the given mapping, preserving this order.
     qubits = circuit.all_qubits()
     key = None
     if mapping:
+        # Add any qubits that were not explicitly mapped, so they aren't lost in
```

Comment:
you could do `key = lambda q: mapping.get(q, q)` if you don't actually need these in `mapping`

Reply:
Nice, done.